extensions/extension.go

```diff
@@ -398,10 +399,11 @@ func jsonSerGo(args []object.Object) object.Object {
 }
 
 func evalFunc(env any, name string, args []object.Object) object.Object {
-	s := args[0].(object.String).Value
-	res, err := eval.EvalString(env, s, name == "unjson" /* empty env */)
+	str := args[0].(object.String).Value
+	s := env.(*eval.State)
+	res, err := eval.EvalString(s, str, name == "unjson" /* empty env */)
 	if err != nil {
-		return object.Error{Value: err.Error()}
+		return s.Error(err.Error())
```

Comment:
This pattern is repeated.

You could use `any` the in s.Error, instead of string, the use fmt.Sprint in s.Error to convert the any into a string, allowing to pass an error

Reply:
I used a big regexp search and replace, then noticed the Sprintf variants and replaced that with Errorf but yeah I might as well take error in too

Reply:
`any` is a bit ugly though as I can then accidentally pass something wrong

Reply:
will add
```go
func (s *State) ErrToError(err error) object.Error {
	return s.Error(err.Error())
}
```
instead